test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp

```diff
@@ -0,0 +1,220 @@
+{
+public:
+
+    ParticipantsDatabase()
+    {
+    }
+
+    ParticipantsDatabase(
+            std::map<ParticipantId, std::shared_ptr<IParticipant>> participants)
+        : ParticipantsDatabase()
+    {
+        participants_ = participants;
+    }
+
+    std::shared_ptr<IParticipant> pop(
+            const ParticipantId& id) noexcept
+    {
+        return pop_(id);
+    }
+
+    std::shared_ptr<IParticipant> pop() noexcept
+    {
+        return pop_();
+    }
+
+    void add_participant(
+            ParticipantId id,
+            std::shared_ptr<IParticipant> participant,
+            std::size_t expected_size) noexcept
+    {
+        add_participant_(id, participant);
+        // Verify correct insertion
+        auto it = participants_.find(id);
+        ASSERT_EQ(it->second, participant);
+        ASSERT_EQ(participants_.size(), expected_size);
+    }
+
+};
+
+} /* namespace test */
+} /* namespace ddsrouter */
+} /* namespace eprosima */
+
+/*********************
+* PROTECTED METHODS **
+*********************/
+
+/**
+ * Test the addition of a participant to the database
+ *
+ * CASES:
+ *
+ *  Add not stored participant
+ *  Add already stored participant
+ */
+TEST(ParticipantsDatabaseTest, add_participant)
+{
+    std::shared_ptr<test::ParticipantsDatabase> participants_database = std::make_shared<test::ParticipantsDatabase>();
+    // Check database is empty
+    ASSERT_TRUE(participants_database->empty());
+
+    ParticipantId id("void_part");
+    std::shared_ptr<VoidParticipant> participant = std::make_shared<VoidParticipant>(id);
+    // Insert participant
+    participants_database->add_participant(participant->id(), participant, 1);
+    // Reinsert to assure there is no error when inserting duplicates, only a warning is printed
+    participants_database->add_participant(participant->id(), participant, 1);
+
+    // Verifications already included in \c add_participant
+}
+
+/**
+ * Test the extraction of a participant from the database
+ *
+ * CASES:
+ *  Empty database
+ *  Pop random participant
+ *  Pop specific participant
+ *  Pop not stored participant
+ *  Pop stored participant
+ */
+TEST(ParticipantsDatabaseTest, pop)
+{
+    std::shared_ptr<test::ParticipantsDatabase> participants_database = std::make_shared<test::ParticipantsDatabase>();
+    ASSERT_TRUE(participants_database->empty());
+    // Empty database, no participant should be popped
+    ASSERT_TRUE(participants_database->pop() == nullptr);
+
+    ParticipantId id("void_part");
+    std::shared_ptr<VoidParticipant> participant = std::make_shared<VoidParticipant>(id);
+    // Pop not yet stored participant
+    ASSERT_TRUE(participants_database->pop(id) == nullptr);
+
+    // Insert participant into database
+    participants_database->add_participant(participant->id(), participant, 1);
+    ASSERT_EQ(participants_database->get_participant(id), participant);
+
+    // Pop stored participant
+    ASSERT_EQ(participants_database->pop(id), participant);
```

Comment:
Test the same but using the `pop()` function.

Reply:
:+1: